jujugui/static/gui/src/app/components/post-deployment/post-deployment.js

```diff
@@ -35,106 +35,48 @@ class PostDeployment extends React.Component {
-      this._getEntityCallback.bind(this));
+    const props = this.props;
+    const getFile = props.charmstore.getFile;
+    props.entityURLs.forEach(entityURL => {
+      const url = urls.URL.fromAnyString(entityURL).toLegacyString();
+      getFile(url, 'getstarted.md', this._handleFileResponse.bind(this, 'content'));
+      getFile(url, 'post-deployment.sh', this._handleFileResponse.bind(this, 'script'));
+    });
   }
 
-  /**
-    Callback handler for getting the entity details.
-
-    @param {Object} error Error from the API.
-    @param {Array} entityData The returned data.
-  */
-  _getEntityCallback(error, entityData) {
+  _handleFileResponse(type, error, fileContents) {
     if (error) {
-      console.error(error);
-      console.error(`Entity not found with id: ${this.props.entityId}`);
+      console.log(error);
       return;
     }
-
-    const files = entityData[0].files;
-    let fileName = 'getstarted.md';
-    if (files && files.some(file => {
-      if (file.toLowerCase() === fileName) {
-        fileName = file;
-        return true;
-      }
-    })) {
-      this.props.charmstore.getFile(
-        this.props.entityId,
-        fileName,
-        this._getGetStartedCallback.bind(this)
-      );
-    }
-    fileName = 'post-deployment.sh';
-    if (files && files.some(file => {
-      if (file.toLowerCase() === fileName) {
-        fileName = file;
-        return true;
-      }
-    })) {
-      this.props.charmstore.getFile(
-        this.props.entityId,
-        fileName,
-        this._getPostDeploymentScriptCallback.bind(this)
-      );
-    }
-  }
-
-  /**
-    Callback when getstarted.md is retrieved.
-    Even if there is an error we still want to display the basic box.
-
-    @param {Object} error Error from the API.
-    @param {String} usageContents The contents of the file.
-  */
-  _getGetStartedCallback(error, usageContents) {
-    if (error) {
-      console.error(error);
-    }
     // Rather then parsing the JSON to read the error we check if the returned
     // body starts with a '{'. If it does, it's not a markdown file so ignore.
-    if (usageContents && usageContents.substring(0, 1) !== '{') {
-      const frontmatterAndMarkdown = this.extractFrontmatter(usageContents);
-      this.setState({
-        metadata: frontmatterAndMarkdown.metadata
-      });
-      let renderer = new marked.Renderer();
-      renderer.link = (href, title, text) => {
-        return `<a href="${href}" title="${title}" target="_blank">${text}</a>`;
-      };
-      const markdown = marked(
-        this.replaceTemplateTags(
-          frontmatterAndMarkdown.markdown
-        ),
-        {renderer: renderer}
-      );
-
-      this.setState({
-        content: markdown
-      });
+    if (fileContents.substring(0, 1) === '{') {
+      return;
     }
+    if (type === 'content') {
+      fileContents = this._processGetStarted(fileContents);
+    }
+    this.setState({[type]: fileContents});
```

Comment:
What happens if there are multiple `entityURLs`?

Reply:
it does....this isn't the final solution but an interim one until we have a real plan for multiple different versions of them.